src/mixins/numeral-formatable.js

```diff
@@ -0,0 +1,278 @@
+      }
+    }
+  }),
+
+  computed: {
+    precision () {
+      return this.absVal(this.options.precision)
+    },
+    positiveSign () {
+      return this.options.positiveSign ? '+' : ''
+    },
+    decimal () {
+      return this.oneChar(this.options.decimal)
+    },
+    blockSeparator () { // Default separator = ' '
+      const separator = this.options.blockSeparator
+
+      if (!Array.isArray(separator)) return [this.oneChar(separator, ' ')]
+
+      switch (separator.length) {
+        case 0: return [' ']
+        case 1: return this.oneChar(separator, ' ')
+        default: return separator
+      }
+    },
+    blockSize () { // Default size = 3
+      const size = this.options.blockSize
+
+      if (!Array.isArray(size)) return [this.absVal(size)]
+
+      switch (size.length) {
+        case 0: return [3]
+        case 1: return [this.absVal(size[0])]
+        default: return size
+      }
+    },
+    sign: {
+      get () {
+        return this.signSymbol
+      },
+      set (val) {
+        let i = 0
+
+        if (this.numeralPrefix) {
+          for (const char of val) {
+            if (allowedMasks['#'].test(char)) break
+            if (char === '+' || char === '-') break
+            i++
+          }
+          val = val.substr(i)
+        }
+
+        const negative = val && val.length > 0 && val[0] === '-'
+        const positiveSymbol = this.positiveSign ? '+' : ''
+
+        this.signSymbol = this.options.positiveOnly ? positiveSymbol
+          : negative ? '-' : positiveSymbol
+      }
+    },
+    numeralPrefix () {
+      return this.options.prefix ? this.options.prefix : ''
+    },
+    numeralSuffix () {
+      return this.options.suffix ? this.options.suffix : ''
+    }
+  },
+
+  methods: {
+    unmaskNumeralText (text) {
+      if (!text) return text
+
+      this.sign = text = String(text)
+      text = this.internalChange ? this.removeNonNumeral(text) : this.attemptNumeralCorrection(text)
+      if (!text) return text // attemptNumeralCorrection() may return null or falsy
+
+      let number
+      if (this.precision) {
+        number = text.split('')
+        number.splice(-this.precision, 0, '.')
+        number = this.parseFloat(number)
+      } else {
+        number = this.parseInt(text)
+      }
+
+      return this.sign + number.toString()
+    },
+    maskNumeralText (text) {
+      if (!text) return this.numeralZero()
+
+      const blockSize = Array.isArray(this.blockSize) ? this.blockSize : [this.blockSize]
+      const masked = []
+
+      let i = this.blockSeparator.length - 1
+      let j = blockSize.length - 1
+      let length = 0
+      let integer = !this.precision
+
+      text = this.sign ? String(text).substr(1) : String(text)
+
+      // Group the digits with specified separators.
+      // The first size and separator element
+      // of the arrays becomes the default value.
+      for (const digit of text.split('').reverse()) {
+        digit !== '.' && masked.push(digit)
```

Comment:
Does it (and 6 lines below) assume that decimal point is always a dot?

Reply:
Yes, it's always a dot in `this.lazyValue`. The algorithm used for calculation is all based on `this.lazyValue`, which holds the actual number.